src/coreclr/jit/gcencode.cpp

```diff
@@ -50,8 +50,21 @@ ReturnKind GCInfo::getReturnKind()
         case 1:
             return VarTypeToReturnKind(retTypeDesc.GetReturnRegType(0));
         case 2:
-            return GetStructReturnKind(VarTypeToReturnKind(retTypeDesc.GetReturnRegType(0)),
-                                       VarTypeToReturnKind(retTypeDesc.GetReturnRegType(1)));
+        {
+            var_types first = retTypeDesc.GetReturnRegType(0);
```

Comment:
This is the right fix for backport. In main, this path is unreachable in main - the whole method can be under `TARGET_X86` ifdef and simplified to only handle a single register.

@jakobbotsch How would you like to handle this?

Reply:
I have pushed a commit with the cleanup